doc/contributing/coding-conventions.chapter.md

```diff
@@ -220,7 +220,9 @@ There are a few naming guidelines:
 
 - The `version` attribute _must_ start with a digit e.g`"0.3.1rc2".
 
-- If a package is not a release but a commit from a repository, then the `version` attribute _must_ be the date of that (fetched) commit. The date _must_ be in `"unstable-YYYY-MM-DD"` format.
+- If a package is a commit from a repository without a version assigned, then the `version` attribute _should_ be the latest upstream version preceding that commit, followed by `-unstable-` and the date of the (fetched) commit. The date _must_ be in `"YYYY-MM-DD"` format.
+
+Example: Given a project had its latest releases `2.2` in November 2021, and `3.0` in January 2022, a commit authored on March 15, 2022 for an upcoming bugfix release `2.2.1` would have `version = "2.2-unstable-2022-03-15"`.
```

Comment:
or 3.0 and to know that for sure you would need to inspect the git tree. If upstream is using multiple branches to track releases that would not be that easy because the commit could be in both, so you would need to browse around the repo to double check things which is another burden for reviewers.

Reply:
It's not like a domain change, there are always small things where we can trust the patch author if everything looks sensible.

Reply:
I am running a lot into situations where authors do not set the version number correct and suggesting the old format is trivial while the new one requires several extra steps I need to take to make sure the suggestion is correct.

Under those conditions I cannot just trust authors to set this correct and I think having wrong information is more harmful than not having that information at all.

Reply:
Let's automate the work, then. It should be fairly straightforward to write a maintainer script that checks the Git repository for the most recent tag preceding a commit and edits the version accordingly, right?

Reply:
A first attempt: #241428